contrib/gorilla/mux/mux.go

```diff
@@ -40,8 +40,12 @@ func (r *Router) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	)
 	// get the resource associated to this request
 	if r.Match(req, &match) {
```

Comment:
I feel there is a lot of `route = "unknown"` in here. Let's simplify this a bit. How about:
```go
route  := "unknown"
if r.Match(req, &match) && match.Route != nil {
    if r, err := match.Route.GetPathTemplate(); err == nil {
        route = r
    }
}
```
I feel this would be simpler.

Lastly, is there any way we can cover this use-case scenario in a test?